### PR TITLE
Open RepositoryObject in VersionService.open

### LINKED ISSUE #4814: Open RepositoryObject in VersionService.open
If a RepositoryObject exists, it should be opened (new RepositoryObjectVersion created with incremented version, cocina copied from last closed version, provided description set). This should be in addition to existing work done in VersionService.open.

Mutating the RepositoryObject should be encapsulated in the CocinaObjectStore. (That is, RepositoryObjects should not be manipulated in VersionService code.)

Please leave a note indicating that the "if a RepositoryObject exists" should be removed. See https://github.com/sul-dlss/dor-services-app/blob/main/app/services/cocina_object_store.rb#L140-L141

### PR DESCRIPTION
**Pull request by jcoyne**
## Why was this change made? 🤔

Fixes #4814

## How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact, including data writes to shared file systems, ***run [integration tests](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test in [stage|qa] environment, in addition to specs. ⚡

### CODE REVIEW

File: spec/services/version_service_spec.rb [hunk 1]
```diff
@@ -23,8 +23,16 @@
     let(:workflow_client) do
       instance_double(Dor::Workflow::Client, create_workflow_by_name: true)
     end
+    # Doing build(:repository_object) rather than create to support test where it doesn't exist.
+    # This case can be removed after we fully migrate to RepositoryObjects
+    let(:repository_object) { build(:repository_object, external_identifier: druid) }
 
     before do
+      # This check be removed after we fully migrate to RepositoryObjects
```
Comment (justinlittman) [inline spec/services/version_service_spec.rb:31]: Can you prefix with TODO so we can find it easier later?

File: app/services/version_service.rb [hunk 2]
```diff
@@ -70,6 +70,13 @@ def open(cocina_object:, description:, opening_user_name:, assume_accessioned:,
       # This is for testing when we don't have the SDR container available
       new_object_version = ObjectVersion.increment_version(druid:, description:)
     end
+
+    # TODO: After migrating to RepositoryObjects, we can get rid of the nil check and use:
+    #   RepositoryObject.find_by!(external_identifier: druid).open_version!
+    RepositoryObject.find_by(external_identifier: druid)&.open_version!
+    # TODO: when we stop calling the UpdateObjectService, after we've migrated to RepostoryObjects, we may need to trigger indexing:
```
Comment (justinlittman) [inline app/services/version_service.rb:77]: I don't think we'll want to stop calling UpdateObjectService, as that service handles all of the activities to update an object, including publishing messages.

**Review by justinlittman — APPROVED** [review]
Approve pending consideration of comments.

File: app/services/version_service.rb (hunk 2 above)
Comment (jcoyne) [reply app/services/version_service.rb:77]: The old way of incrementing a version was to persist with an update. Now, open_object! does the persistence. I don't think we want a duplicate update message. Just the reindex should be sufficient.

**Pull request merged**